### PR TITLE
Query Results Page

### PR DESCRIPTION
**Pull request by jtr-usaf**
image

### CODE REVIEW

File: src/app/routes/query.py [hunk 1]
```diff
@@ -21,32 +20,69 @@ def route():
 
 
 def route_results():
-    if request.method == 'POST':
-        query_name = request.form.get('query_name') or uuid4()
-        query_text = request.form.get('query')
-        selected_auths = [auth for auth in request.form.getlist('auths') if auth]
+    # We need to handle our args in a few different ways, first and formost if they are passed as args in the url
```
Comment (DL-USAF) [inline src/app/routes/query.py:23]:
```suggestion
    # We need to handle our args in a few different ways, first and foremost if they are passed as args in the url
```

File: src/app/routes/query.py [hunk 2]
```diff
@@ -21,32 +20,69 @@ def route():
...
+        return render_template('query_result.html', no_query=True)
 
-        if not selected_auths:
-            flash('Please select at least one Auth option.', 'error')
-            return redirect(url_for('main.query'))
+    # Extract our arguments
+    query_name = arg_source.get('query_name') or uuid4()
+    query = arg_source.get('query')
+    data_type = arg_source.get('data_type')
+    decode_raw_data = bool(arg_source.get('decode_raw_data'))
```
Comment (DL-USAF) [inline src/app/routes/query.py:44]: add a todo mentioning we need to add the other query parameters as options.

File: src/app/routes/query.py [hunk 3]
```diff
@@ -21,32 +20,69 @@ def route():
...
+    if request.method == 'POST':
+        # POST requests are redirected to a GET with URL arguments so the users can copy their query URL if they want
+        # If they want a json response we need to pass that along by using the `.json` endpoint
+        endpoint = 'main.query_results' + ('_json' if user_requests_json(request) else '')
+        return redirect(url_for(endpoint, query=query, auths=selected_auths,
+                                query_name=query_name, data_type=data_type, decode_raw_data=decode_raw_data))
+    elif request.method == 'GET':
+        # GET requests execute the query
         output_location = f'/tmp/{query_name}/results.json'
         cmd = [
```
Comment (DL-USAF) [inline src/app/routes/query.py:61]: add a todo mentioning we will need to update the cert values, just so that doesn't get forgotten about.

**Review by DL-USAF — CHANGES_REQUESTED** [review]
see comments on code

File: src/app/routes/query.py (hunk 2 above)
Comment (jtr-usaf) [reply src/app/routes/query.py:44]: Added a todo right after this.

File: src/app/routes/query.py (hunk 3 above)
Comment (jtr-usaf) [reply src/app/routes/query.py:61]: Added TODOs for all occurrences of [PLACEHOLDER] for cert values.

**Review by DL-USAF — APPROVED** [review]

**Pull request merged**